pkg/controller/specialresource/specialresource_controls.go

```diff
@@ -0,0 +1,544 @@
+		return NotReady, err
+	}
+
+	logger.Info("Found")
+
+	return Ready, nil
+}
+
+func ClusterRoleBinding(n SRO) (ResourceStatus, error) {
+
+	state := n.idx
+	obj := &n.resources[state].ClusterRoleBinding
+
+	found := &rbacv1.ClusterRoleBinding{}
+	logger := log.WithValues("ClusterRoleBinding", obj.Name, "Namespace", obj.Namespace)
+
+	if err := controllerutil.SetControllerReference(n.ins, obj, n.rec.scheme); err != nil {
+		return NotReady, err
+	}
+
+	logger.Info("Looking for")
+	err := n.rec.client.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: obj.Name}, found)
+	if err != nil && errors.IsNotFound(err) {
+		logger.Info("Not found, creating")
+		err = n.rec.client.Create(context.TODO(), obj)
+		if err != nil {
+			logger.Info("Couldn't create", "Error", err)
+			return NotReady, err
+		}
+		return Ready, nil
+	} else if err != nil {
+		return NotReady, err
+	}
+
+	logger.Info("Found")
+
+	return Ready, nil
+}
+
+func ConfigMap(n SRO) (ResourceStatus, error) {
+
+	state := n.idx
+	obj := &n.resources[state].ConfigMap
+
+	found := &corev1.ConfigMap{}
+	logger := log.WithValues("ConfigMap", obj.Name, "Namespace", obj.Namespace)
+
+	if err := controllerutil.SetControllerReference(n.ins, obj, n.rec.scheme); err != nil {
+		return NotReady, err
+	}
+
+	logger.Info("Looking for")
+	err := n.rec.client.Get(context.TODO(), types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, found)
+	if err != nil && errors.IsNotFound(err) {
+		logger.Info("Not found, creating")
+		err = n.rec.client.Create(context.TODO(), obj)
+		if err != nil {
+			logger.Info("Couldn't create", "Error", err)
+			return NotReady, err
+		}
+		return Ready, nil
+	} else if err != nil {
+		return NotReady, err
+	}
+
+	logger.Info("Found")
+
+	return Ready, nil
+}
+
+func kernelFullVersion(n SRO) string {
+
+	logger := log.WithValues("Request.Namespace", "default", "Request.Name", "Node")
+	// We need the node labels to fetch the correct container
+	opts := &client.ListOptions{}
+	opts.SetLabelSelector("feature.node.kubernetes.io/pci-10de.present=true")
+	list := &corev1.NodeList{}
+	err := n.rec.client.List(context.TODO(), opts, list)
+	if err != nil {
+		logger.Info("Could not get NodeList", err)
+	}
+	// Assuming all nodes are running the same kernel version,
+	// One could easily add driver-kernel-versions for each node.
+	node := list.Items[0]
+	labels := node.GetLabels()
+
+	var ok bool
+	kernelFullVersion, ok := labels["feature.node.kubernetes.io/kernel-version.full"]
+	if ok {
+		logger.Info(kernelFullVersion)
+	} else {
+		err := errors.NewNotFound(schema.GroupResource{Group: "Node", Resource: "Label"},
+			"feature.node.kubernetes.io/kernel-version.full")
+		logger.Info("Couldn't get kernelVersion", err)
+		return ""
+	}
+	return kernelFullVersion
+}
+
+func preProcessDaemonSet(obj *appsv1.DaemonSet, n SRO) {
+	if obj.Name == "nvidia-driver-daemonset" {
+		kernelVersion := kernelFullVersion(n)
+		img := obj.Spec.Template.Spec.Containers[0].Image
```

Comment:
I am not sure about what happens here if `kernelFullVersion` return `""` (in the case the NFD label is not present). Could you give us some details?